### PR TITLE
If an agent has bookings the change-agency request return 400

### PR DESCRIPTION
**Pull request by AlexeyShok**
If an agent has bookings, the change-agency request now return 400. Accordingly, the transfer does not take place.
If the source agency has no more agents, it is deactivated. If there are no more active agencies in the source counterparty, it is deactivated.

### CODE REVIEW

File: Api/Services/Management/AgentMovementService.cs [hunk 1]
```diff
@@ -12,18 +13,24 @@ namespace HappyTravel.Edo.Api.Services.Management
...
         
         
         public async Task<Result> Move(int agentId, int sourceAgencyId, int targetAgencyId, List<int> roleIds)
         {
             return await ValidateRequest()
+                .Ensure(AgentHasNoBookings, $"Agent {agentId} has bookings in agency {sourceAgencyId}")
                 .Bind(UpdateAgencyRelation)
-                .Bind(WriteLog);
+                .Bind(WriteLog)
+                .Bind(DeactivateAgencyIfNeed)
```
Comment (mingazhev) [inline Api/Services/Management/AgentMovementService.cs:32]: If needed

File: Api/Services/Management/AgentMovementService.cs [hunk 2]
```diff
@@ -12,18 +13,24 @@ namespace HappyTravel.Edo.Api.Services.Management
...
+                .Bind(DeactivateCounterpartyIfNeed);
```
Comment (mingazhev) [inline Api/Services/Management/AgentMovementService.cs:33]: If needed

File: Api/Services/Management/AgentMovementService.cs [hunk 3]
```diff
@@ -40,6 +47,14 @@ async Task<Result> ValidateRequest()
             }
 
 
+            async Task<bool> AgentHasNoBookings()
+            {
+                var firstBooking = await _edoContext.Bookings.FirstOrDefaultAsync(b => b.AgentId == agentId && b.AgencyId == sourceAgencyId);
```
Comment (mingazhev) [inline Api/Services/Management/AgentMovementService.cs:52]: AnyAsync will be more short and clear there

File: Api/Services/Management/AgentMovementService.cs (hunk 1 above)
Comment (AlexeyShok) [reply Api/Services/Management/AgentMovementService.cs:32]: Fixed

File: Api/Services/Management/AgentMovementService.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/Services/Management/AgentMovementService.cs:33]: Fixed

File: Api/Services/Management/AgentMovementService.cs (hunk 3 above)
Comment (AlexeyShok) [reply Api/Services/Management/AgentMovementService.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**